jcl/src/java.base/share/classes/java/lang/String.java

```diff
@@ -2720,9 +2720,17 @@ public String toLowerCase(Locale locale) {
 		}
 
 		if (StrHWAvailable() && language == "en") { //$NON-NLS-1$
-			String output = new String(lengthInternal());
-			if (toLowerHWOptimized(output))
-				return output;
+			int sLength = lengthInternal();
+
+			if (enableCompression && (null == compressionFlag || coder == LATIN1)) {
+				byte[] output = new byte[sLength];
+				if (toLowerHWOptimizedLatin1(value, output, sLength))
```

Comment:
Perhaps we can match the new name of the check and rename these accordingly to `toLowerIntrinsicLatin1` / `toLowerIntrinsicUTF16` (similarly for `toUpper`).